packages/react-components/src/components/Drawer/DrawerMenuItem.tsx

```diff
@@ -0,0 +1,76 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { Icon, IIconProps, IStyle, mergeStyles, Stack } from '@fluentui/react';
+import React from 'react';
+import { useTheme } from '../../theming/FluentThemeProvider';
+import { BaseCustomStyles } from '../../types';
+import { submitWithKeyboard } from '../utils/keyboardNavigation';
+
+/**
+ * Props for the DrawerMenuItem
+ *
+ * @internal
+ */
+export interface _DrawerMenuItemProps {
+  onItemClick?: () => void;
+  key: string;
+  text?: string;
+  iconProps?: IIconProps;
+  styles?: BaseCustomStyles;
+
+  // Support for submenu coming in follow up PR
+  // subMenuProps?: _DrawerMenuItemProps[];
+}
+
+/**
+ * Maps the individual item in menuProps.items passed in the {@link DrawerMenu} into a UI component.
+ *
+ * @private
+ */
+export const DrawerMenuItem = (props: _DrawerMenuItemProps): JSX.Element => {
+  const palette = useTheme().palette;
+  const rootStyles = mergeStyles(drawerMenuItemRootStyles(palette.neutralLight), props.styles?.root);
+  const onClick = (): void => props.onItemClick && props.onItemClick();
```

Comment:
nit: I'd inline this.

Reply:
I updated to be used by the `onKeyPress` (in higher PR) so will leave the onClick here

Reply:
Updated in this PR to highlight, moved rootStyles inline